Control/DeepSeq.hs

```diff
@@ -750,6 +795,16 @@ instance NFData (FunPtr a) where
 instance NFData1 FunPtr where
     liftRnf _ = rwhnf
 
+----------------------------------------------------------------------------
+-- Foreign.ForeignPtr
+
+-- |@since 1.4.5.0
```

Comment:
I think these Haddocks also deserve a `__NOTE__: Only strict in the reference and not the referenced value.` comment, given your observation that this instance is strict in the `ForeignPtr`'s `Addr#` but not its `ForeignPtrContents`.